gensim/models/word2vec_inner.pyx

```diff
@@ -709,21 +719,22 @@ def score_sentence_sg(model, sentence, _work):
     cdef long result = 0
     cdef int sentence_len
 
-    c.syn1 = <REAL_t *>(np.PyArray_DATA(model.trainables.syn1))
+    c.syn1 = <REAL_t *>(np.PyArray_DATA(model.syn1))
 
     # convert Python structures to primitive types, so we can release the GIL
     c.work = <REAL_t *>np.PyArray_DATA(_work)
 
-    vlookup = model.wv.vocab
+    vocab_codes = model.wv.expandos['code']
+    vocab_points = model.wv.expandos['point']
     i = 0
     for token in sentence:
-        word = vlookup[token] if token in vlookup else None
-        if word is None:
-            continue  # should drop the
-        c.indexes[i] = word.index
-        c.codelens[i] = <int>len(word.code)
-        c.codes[i] = <np.uint8_t *>np.PyArray_DATA(word.code)
-        c.points[i] = <np.uint32_t *>np.PyArray_DATA(word.point)
+        word_index = model.wv.key_to_index[token] if token in model.wv.key_to_index else None
+        if word_index is None:
+            continue  # for score, should this be a default negative value?
```

Comment:
What do you mean?

Reply:
These 'score' functions are a long-ago contribution from @mataddy whose current function/utility is unclear. I've continued to apply mechanical updates to match other changes, and the code still compiles & passes the one (trivial, form-but-not-function) unit test. But it's an idiosyncratic technique, and only works for the non-default `hs` mode. Here, in lieu of the previous cryptic `# should drop the` comment, I've asked if for the purposes of this particular kind of 'scoring' (really, loss-tallying indicating how divergent this new text is from what the model learned during training), shouldn't completely missing words imply something very negative, as opposed to nothing-at-all? But probably, this functionality should be dropped. (And ultimately, a talented cleanup of the largely-broken loss-tallying functions might provide a cleaner window into this same measure of how well a text contrasts with model expectations - such as a way to report loss from a single invocation of one fo the inner train methods, without changing the model.)